toxics/toxic.go

```diff
@@ -32,14 +32,28 @@ type BufferedToxic interface {
 }
 
 type ToxicWrapper struct {
-	Toxic      `json:"-"`
-	Name       string           `json:"name"`
-	Type       string           `json:"type"`
-	Stream     string           `json:"stream"`
-	Toxicity   float32          `json:"toxicity"`
-	Direction  stream.Direction `json:"-"`
-	Index      int              `json:"-"`
-	BufferSize int              `json:"-"`
+	Toxic
+	Name       string
+	Type       string
+	Stream     string
+	Toxicity   float32
+	Direction  stream.Direction
+	Index      int
+	BufferSize int
+}
+
+// Returns a flattened map of the toxic for use with json
+func (w *ToxicWrapper) GetMap() map[string]interface{} {
```

Comment:
The problem is these fields are different for every toxic type. And since it's an interface, the json marshaller can't seem to figure out how to do nested structs. I'm really not happy with the usage of `reflect` either, so if you've got some other Go tricks up your sleeve to show me, I'd love to see them.